io/plugins/eu.esdihumboldt.hale.io.csv/src/eu/esdihumboldt/hale/io/csv/writer/internal/CSVInstanceWriter.java

```diff
@@ -146,10 +145,13 @@ public boolean isPassthrough() {
 	}
 
 	// write current instance to csv file
-	private void writeLine(boolean solveNestedProperties, List<String> headerRow,
-			Instance instance, CSVWriter writer) {
+	private void writeLine(boolean solveNestedProperties, List<String> headerRow, Instance instance,
+			CSVWriter writer) {
+		boolean useSchema = getParameter(InstanceTableIOConstants.USE_SCHEMA).as(Boolean.class,
```

Comment:
Should be an optional parameter to be backwards compatible.

Though as we said the original behavior is a bug, I guess the default should then be `true`.

Remember that when you add a new parameter for an I/O provider you also need to define it in the related extension in `plugin.xml`.

Reply:
Where can I find the right plugin.xml? I have explored some directories but I am not sure I have found the right one. I need to note it down as a recurrent step ;-)

Reply:
The bundle where the `CSVInstanceWriter` is defined should also hold the declaration of it as I/O provider in the extensions.

Reply:
I would then insert the new parameter in this section of the plugin:
```
<provider
            class="eu.esdihumboldt.hale.io.csv.writer.internal.CSVInstanceWriter"
            id="eu.esdihumboldt.hale.io.csv.writer.instance"
            name="CSV file">
         <contentType
               ref="eu.esdihumboldt.hale.io.csv">
         </contentType>
         <providerParameter
               description="Schema type which instances should be exported."
               label="Export type"
               name="selectedExportType"
               optional="false"
               valueDescriptor="eu.esdihumboldt.hale.io.csv.TypenameParameterDescriptor">
            <parameterBinding
                  class="java.lang.String">
            </parameterBinding>
         </providerParameter>
         <providerParameter
               description="Separator character that delimits columns in the CSV file"
               label="Separator character"
               name="separator"
               optional="true">
            <parameterBinding
                  class="java.lang.String">
            </parameterBinding>
            <valueDescriptor
                  defaultDescription="Defaults to tabulator (\t)">
            </valueDescriptor>
         </providerParameter>
         <providerParameter
               description="Character used to quote CSV entries"
               label="Quote character"
               name="quote"
               optional="true">
            <parameterBinding
                  class="java.lang.String">
            </parameterBinding>
            <valueDescriptor
                  default="&quot;">
            </valueDescriptor>
         </providerParameter>
         <providerParameter
               description="Character used to escape special characters, e.g. the quote character"
               label="Escape character"
               name="escape"
               optional="true">
            <parameterBinding
                  class="java.lang.String">
            </parameterBinding>
            <valueDescriptor
                  default="\">
            </valueDescriptor>
         </providerParameter>
      </provider>
   </extension>
```
but why there is no `solveNestedProperties` in here too?

Reply:
> Should be an optional parameter to be backwards compatible.

This is still true. You should use `getOptionalParameter(...).

Reply:
I'm sorry, I mixed that up with the transformation functions, fine as it is.